root/usr/share/fdi/facts/openlldp.rb

```diff
@@ -66,7 +63,7 @@ def grep_v(cond)
   }
 
   # Remove interfaces that pollute the list (like lo and bond0).
-  Facter.value('interfaces').split(/,/).grep_v(/^lo$|^bond[0-9]/).each do |interface|
+  Dir.glob("/sys/class/net/*").map{|x| Pathname.new(x).basename.to_s}.grep_v(/^lo$|^bond[0-9]/).each do |interface|
```

Comment:
Any reason for `Pathname.new(x).basename.to_s` over `File.basename(x)`?

Reply:
Fixed thanks.